pkg/controller/controller.go

```diff
@@ -444,21 +536,129 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 	}, nil
 }
 
+// createMetroVolumeNameSuffix generates the suffix for the name of a metro volume.
+// localResourceID is either a volume ID or a volume group ID and will be used to get the replication session.
+// The replication session is then used to find the remote volume ID corresponding to the local volume ID, volID.
+func createMetroVolumeNameSuffix(ctx context.Context,
+	localResourceID string,
+	localVolID string,
+	remoteSystem gopowerstore.RemoteSystem,
+	arr *array.PowerStoreArray,
+) (string, error) {
+	// Use the replication session to get the remote volume ID
+	replicationSession, err := arr.GetClient().GetReplicationSessionByLocalResourceID(ctx, localResourceID)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, "could not get metro replication session: %s", err.Error())
+	}
+	// Confirm the replication session is of 'volume' or 'volume_group' type
+	resourceType := strings.ToLower(replicationSession.ResourceType)
+	if resourceType != "volume" && resourceType != "volume_group" {
+		return "", status.Errorf(codes.FailedPrecondition, "replication session %s has a resource type %s, wanted type 'volume' or 'volume_group'",
+			replicationSession.ID, replicationSession.ResourceType)
+	}
+
+	// Get the remote volume ID
+	remoteVolumeID := ""
+	if localResourceID == localVolID {
+		remoteVolumeID = replicationSession.RemoteResourceID
+	} else {
+		for _, volumePair := range replicationSession.StorageElementPairs {
+			if volumePair.LocalStorageElementID == localVolID {
+				remoteVolumeID = volumePair.RemoteStorageElementID
```

Comment:
We could break the loop on successful find.

Reply:
Ohp. Yep. will do.